src/proper_target.erl

```diff
@@ -151,57 +245,76 @@ strategy(Strat) ->
       Strat
   end.
 
-%% store the used strategy into the process dictionary
-%% used only to provide backwards compatibility
-%% @private
--spec use_strategy(strategy(), proper:setup_opts()) -> proper:outer_test().
-use_strategy(Strat, Opts) ->
-  Strategy = strategy(Strat),
-  put('$search_strategy', Strategy),
-  Strategy:init_strategy(Opts).
 
--spec init_strategy(strategy()) -> ok.
-init_strategy(Strat) ->
-  Strategy = strategy(Strat),
-  put('$search_strategy', Strategy),
-  Steps = get('$search_steps'),
-  OutputFun = fun(_, _) -> ok end,
-  Strategy:init_strategy(#{numtests=>Steps, output_fun=>OutputFun}).
+%% -----------------------------------------------------------------------------
+%% gen_server callbacks
+%% -----------------------------------------------------------------------------
 
 %% @private
--spec cleanup_strategy() -> ok.
-cleanup_strategy() ->
-  (strategy()):cleanup(),
-  ok.
+-spec init(Args :: [{strategy(), strategy_data()}]) -> {ok, state()}.
+init([{Strategy, Data}]) ->
+  {ok, #state{strategy = Strategy, data = Data}}.
 
 %% @private
--spec get_target(tmap()) -> target().
-get_target(TMap) ->
-  Strategy = strategy(),
-  case Strategy:retrieve_target() of
-    undefined ->
-      FreshTarget = Strategy:init_target(TMap),
-      Strategy:store_target(FreshTarget),
-      FreshTarget;
-    StoredTarget ->
-      StoredTarget
-  end.
+-spec handle_call(Request :: term(), From :: {pid(), Tag :: term()},
+                  State :: state()) ->
+  {reply, Reply :: term(), NewState :: state()}.
+handle_call(gen, _From, State) ->
+  Strat = State#state.strategy,
+  Target = State#state.target,
+  Data = State#state.data,
+  {NextValue, NewTarget, NewData} = Strat:next(Target, Data),
+  {reply, NextValue, State#state{target = NewTarget, data = NewData}};
+handle_call({init_target, TMap}, _From, State) ->
+  Strat = State#state.strategy,
+  Target = State#state.target,
+  NewTarget = case Target of
```

Comment:
The indentation of this `case` is not what you would get by Emacs...   Is this a problem?

Reply:
I think the new identation is correct now.